src/Type/Constant/ConstantArrayType.php

```diff
@@ -303,6 +303,10 @@ public function acceptsWithReason(Type $type, bool $strictTypes): AcceptsResult
 			return AcceptsResult::createFromBoolean(count($type->keyTypes) === 0);
 		}
 
+		if ($this->isList()->yes() && $type->isList()->no()) {
+			return AcceptsResult::createNo();
```

Comment:
What about `$type->isList()->maybe()`? I think that at that point the `$result` below should be assigned to `createMaybe()` and not `createYes()`

Reply:
I added https://github.com/phpstan/phpstan-src/pull/3381/commits/102b53b239123b4f560282c37fdcea44e144ce56

but what about the following issue https://github.com/phpstan/phpstan-src/pull/3381#issuecomment-2361360419 ?

Reply:
I don't really know what to do about that. Maybe this whole change is a hack and we need to approach it differently.

Reply:
I think your right and this things is a hack, or at least not the right fix because it's not really related to list.

PHPStan sometimes consider an array is ordered and sometimes not.

An easy reproducer is https://phpstan.org/r/1ffe9cb5-65b1-4a7c-9010-3aa24560b462 vs real world example https://3v4l.org/TqTBn

Maybe `ConstantArray::getKeysOrValuesArray` is wrong... or a `ordered-array` (or `unordered-array`) need to be introduced...